dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp

```diff
@@ -256,40 +281,40 @@ DataInterfaceIIDM::initFromIIDM() {
   //===========================
   //  ADD 2WTFO INTERFACE
   //===========================
-  for (auto& twoWindingTransfoIIDM : networkIIDM_.getTwoWindingsTransformers()) {
-    shared_ptr<TwoWTransformerInterface> tfo = importTwoWindingsTransformer(twoWindingTransfoIIDM);
+  for (auto& twoWindingTransfoIIDM : networkIIDM_->getTwoWindingsTransformers()) {
+    auto tfo = importTwoWindingsTransformer(twoWindingTransfoIIDM);
     network_->addTwoWTransformer(tfo);
     components_[tfo->getID()] = tfo;
   }
 
   //==========================================
   //  CONVERT THREE WINDINGS TRANSFORMERS
   //==========================================
-  for (auto& threeWindingTransfoIIDM : networkIIDM_.getThreeWindingsTransformers()) {
+  for (auto& threeWindingTransfoIIDM : networkIIDM_->getThreeWindingsTransformers()) {
     convertThreeWindingsTransformers(threeWindingTransfoIIDM);
   }
 
   //===========================
   //  ADD LINE INTERFACE
   //===========================
-  for (auto& lineIIDM : networkIIDM_.getLines()) {
-    shared_ptr<LineInterface> line = importLine(lineIIDM);
+  for (auto& lineIIDM : networkIIDM_->getLines()) {
+    auto line = importLine(lineIIDM);
     network_->addLine(line);
     components_[line->getID()] = line;
   }
 
   //===========================
   //  ADD HVDC LINE INTERFACE
   //===========================
-  for (auto& hvdcLineIIDM : networkIIDM_.getHvdcLines()) {
-    shared_ptr<HvdcLineInterface> hvdc = importHvdcLine(hvdcLineIIDM);
+  for (auto& hvdcLineIIDM : networkIIDM_->getHvdcLines()) {
+    auto hvdc = importHvdcLine(hvdcLineIIDM);
     network_->addHvdcLine(hvdc);
     components_[hvdc->getID()] = hvdc;
   }
   DYNErrorQueue::instance().flush();
 }
 
-shared_ptr<VoltageLevelInterface>
+shared_ptr<VoltageLevelInterfaceIIDM>
```

Comment:
why now use the child class and not the interface ?

Reply:
as it makes useless dynamic cast whereas we know at this point the accurate class

Reply:
so this is not a change that was required by this dev ? and why not do it for the old implem ?

Reply:
It is used to avoidsome dynamic cast in this dev, the old  implementation is supposed to be deleted at short/mid term so no there is no need to report code quality modifications into it